pkg/compose/run.go

```diff
@@ -115,22 +118,18 @@ func (s *composeService) runInteractive(ctx context.Context, containerID string,
 	}
 }
 
-func (s *composeService) terminateRun(ctx context.Context, containerID string, opts api.RunOptions, err error) (int, error) {
-	if err != nil {
-		return 0, err
-	}
-	inspect, err := s.apiClient.ContainerInspect(ctx, containerID)
-	if err != nil {
-		return 0, err
-	}
-	exitCode := 0
-	if inspect.State != nil {
-		exitCode = inspect.State.ExitCode
+func (s *composeService) terminateRun(ctx context.Context, containerID string, opts api.RunOptions) (exitCode int, err error) {
+	exitCh, errCh := s.apiClient.ContainerWait(ctx, containerID, container.WaitConditionNotRunning)
+	select {
+	case exit := <-exitCh:
+		exitCode = int(exit.StatusCode)
+	case err = <-errCh:
+		return
 	}
 	if opts.AutoRemove {
 		err = s.apiClient.ContainerRemove(ctx, containerID, moby.ContainerRemoveOptions{})
 	}
-	return exitCode, err
+	return
```

Comment:
I feel like I'm misreading, but `exitCode` being not being returned in this function?

Reply:
function is using named return variables `(exitCode int, err error)`
so just need to set value then return

Reply:
ah! Thanks!